datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -2462,6 +2470,13 @@ private void registerGauge(String metricName, Supplier<?> valueSupplier) {
       _metricInfos.add(new BrooklinGaugeInfo(_coordinator.buildMetricName(MODULE, metricName)));
     }
 
+    // registers a new gauge or updates the supplier for the gauge if it already exists
+    private <T> void registerOrSetGauge(String metricName, Supplier<T> valueSupplier) {
+      _dynamicMetricsManager.setGauge(_dynamicMetricsManager.registerGauge(MODULE, metricName, valueSupplier),
```

Comment:
Will this metric get emitted for zero value as well?

This can be a concern if enabled for the Change capture cluster with many datastreams.  It will make sense to have `aggregate` level metrics and selectively enable `datastream-level` metrics because X datastream means X Gauge.

Reply:
This will be only be emitted wherever the feature of handling bad actors is enabled. 
Also I updated the logic to only handle and register/update a gauge when this metadata field exists for that datastream.